src/MyConnection.h

```diff
@@ -30,15 +35,21 @@ class MyConnection : boost::noncopyable {
     mysql_options(pConn_, MYSQL_OPT_LOCAL_INFILE, 0);
     // Default to UTF-8
     mysql_options(pConn_, MYSQL_SET_CHARSET_NAME, "UTF8");
-    if (groups != "")
-      mysql_options(pConn_, MYSQL_READ_DEFAULT_GROUP, groups.c_str());
-    if (default_file != "")
-      mysql_options(pConn_, MYSQL_READ_DEFAULT_FILE, default_file.c_str());
-
-
-    if (!mysql_real_connect(pConn_, host.c_str(), user.c_str(),
-        password.c_str(), db == "" ? NULL : db.c_str(), port,
-        unix_socket == "" ? NULL : unix_socket.c_str(), client_flag)) {
+    if (!groups.isNull())
+      mysql_options(pConn_, MYSQL_READ_DEFAULT_GROUP,
+                    Rcpp::as<std::string>(groups).c_str());
```

Comment:
I don't know. This is going over my head a bit, and costing me a lot of time.

Could you perhaps reconsider? I mean, we now see that this is not "a small function" anymore. Compared to this, the original `host.isNull() ? NULL : Rcpp::as<std::string>(host).c_str()` doesn't look so bad to me. The pattern is only repeated five times in total, quite clearly arranged in the call to `mysql_real_connect()`. Of course, there are also those two calls to `mysql_options()` that look a bit similar, but I think it would be unnecessary and a bit unpredictable to make those calls with `arg=NULL`, so those wouldn't benefit from `as_cstr()`.

And since you authored the entire `as_cstr()` class already now, I feel it's not my commit to make anyway.

How about I create a new branch, with the const reference additions of last week, I send you a new pull request, and you pull it in as a new working branch here? That way you can add your own commits to mine and merge the entire branch with master as soon as the new Rcpp comes out.